src/planscape/impacts/services.py

```diff
@@ -528,3 +528,58 @@ def calculate_project_area_deltas(
         }
     )
     return results
+
+
+from impacts.models import TreatmentResult, ImpactVariable
+
+
+def get_treatment_results_table_data(treatment_plan, stand_id):
+    """
+    Retrieves treatment results for RATE_OF_SPREAD and FLAME_LENGTH for the given stand and treatment plan.
+    Return a list of dictionaries: [{"year": ..., "rate_of_spread": ..., "flame_length": ...}, ...]
+    """
+    # Filter results for the given stand and plan, for the two variables we need
+    queryset = TreatmentResult.objects.filter(
+        treatment_plan=treatment_plan,
+        stand_id=stand_id,
+        variable__in=[ImpactVariable.RATE_OF_SPREAD, ImpactVariable.FLAME_LENGTH],
+    )
```

Comment:
You can let the front-end decide what needs to be filtered.

Also, you can simplify it by returning one object per result. Ex: `[{"year": 2024, "variable": "FL", "value": "High"},...]`
This way, you can execute a query instead of iterate over the result and reducing the number of queries.
```suggestion
    Retrieves treatment results for RATE_OF_SPREAD and FLAME_LENGTH for the given stand and treatment plan.
    Return a list of dictionaries: [{"year": ..., "rate_of_spread": ..., "flame_length": ...}, ...]
    """
    filters = {}
    if variables:
        filters = {"variable__in": variables}
    # Filter results for the given stand and plan, for the two variables we need
    queryset = TreatmentResult.objects.filter(
        treatment_plan=treatment_plan,
        stand_id=stand_id,
        **filters
    )
    result = queryset.values("year", "variable", "value") 
```

Reply:
Also, another way, and probably better, is to return the filtered queryset and build the return with a serializer in the viewset.